src/Sarif.Viewer.VisualStudio/Options/SarifViewerOptionPage.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.Sarif.Viewer.Options
+{
+    [ComVisible(true)]
+    public class SarifViewerOptionPage : DialogPage
```

Comment:
>SarifViewerOptionPage [](start = 17, length = 21)

imagine that it started as true.
when u change to false, does it trigger any event to cancel previous watchers?

Reply:
Option value was checked in StartWatch/StopWatch methods, it only happens  when solution is loaded/closed.
If option changed, the change would effect next time solution is loaded/closed, before that any file added/deleted still be watched.
In latest change moved the checking to from Start/StopWatch to file created/deleted events, so the change will effect immediately. 

---
In reply to: [598008582](https://github.com/microsoft/sarif-visualstudio-extension/pull/382#discussion_r598008582) [](ancestors = 598008582)